app/components/build/AnnotationsList.js

```diff
@@ -43,6 +42,49 @@ class AnnnotationsList extends React.Component {
     );
   }
 
+  handleAnnotationClick = (event: MouseEvent) => {
+    // Don't change anything if the user is using any modifier keys
+    if ((event.button !== 0) || event.altKey || event.ctrlKey || event.metaKey || event.shiftKey) {
+      return;
+    }
+
+    // Don't change clicks in things which aren't <a /> elements, or don't have a `href`
+    if (!(event.target instanceof HTMLAnchorElement) || !event.target.href) {
+      return;
+    }
+
+    // Parse the link URL
+    const linkUrl = new URL(event.target.href);
```

Comment:
Any chance this can raise an error?

Reply:
If event.target.href is not a valid url, yeah, it’ll raise a TypeError. I guess if it’s not valid this should just, what, continue as normal?

Reply:
I can’t actually find a way to provide an “invalid” URL to this, given that all the URLs are normalised through both the annotation gear _and_ the browser.

If an error _does_ get thrown in this handler, the browser will navigate as normal (and the user will likely get a 404, lol), and we might even get it in Bugsnag, if we’re lucky. I’d love to know how it’s even possible!

Reply:
I’ve also added an extra layer of insurance against this in b8720de; it provides the current URL as the base to the URL constructor, so even if the browser’s normalisation doesn’t mean we get a fully-qualified URL from the DOM (unlikely, but older browsers didn’t do this right), we’ll always produce what the browser interprets it as, which means domain-relative URLs will always be valid.

Reply:
Perfect! I don't even think it's a serious issue, just something I thought of.